src/main/java/com/google/api/generator/gapic/protoparser/Parser.java

```diff
@@ -641,10 +640,51 @@ static LongrunningOperation parseLro(
     OperationInfo lroInfo =
         methodDescriptor.getOptions().getExtension(OperationsProto.operationInfo);
 
-    String responseTypeName = parseNestedProtoTypeName(lroInfo.getResponseType());
-    String metadataTypeName = parseNestedProtoTypeName(lroInfo.getMetadataType());
-    Message responseMessage = messageTypes.get(responseTypeName);
-    Message metadataMessage = messageTypes.get(metadataTypeName);
+    // These can be short names (e.g. FooMessage) or fully-qualified names with the *proto* package.
+    String responseTypeName = lroInfo.getResponseType();
+    String metadataTypeName = lroInfo.getMetadataType();
+    String[] responseTypeNameComponents = responseTypeName.split("\\.");
```

Comment:
Minor, but it seems like you don't need the full set of compontents, only the last element after dot, so it can be something like.
```
String responseTypeShortName = responseTypeName.substring(responseTypeName.lastIndexOf(".") + 1)
```
It is not about performance, mainly just to make it clear that the other compnents are not needed to the readers of the code.

Reply:
Done.